### PR TITLE
Major performance improvements

### LINKED ISSUE #8: Improve performance for large lists
Searching and loading large (~300 or more) lists of attendees causes a significant delay that is only made worse by the fact that the list is destroyed and recreated upon every keypress in the search field. [Relevant section of code](https://github.com/HackGT/Ultimate-Checkin/blob/master/client/js/main.ts#L136) that removes and recreates new DOM elements in an inefficient manner.

### PR DESCRIPTION
**Pull request by petschekr**
- Moves some Cheerio-based hacky rendering to Handlebar templates
- Simplifies loading of states and implements returning to the same state based on the URL hash
- Greatly improves performance when searching through large lists of attendees by preallocating list elements from the template when the page is first loaded and reusing them instead of deleting and regenerating all of them on each new search.
    - Testing on a list of 500 attendees yielded an improvement of ~15x in the best case.

Closes #8

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,11 +32,11 @@
...
     "@types/body-parser": "0.0.33",
     "@types/chai": "^3.4.34",
-    "@types/cheerio": "^0.17.31",
     "@types/compression": "0.0.33",
     "@types/cookie-parser": "^1.3.30",
     "@types/csv-parse": "^1.1.6",
     "@types/express": "^4.0.34",
+    "@types/handlebars": "^4.0.32",
     "@types/mocha": "^2.2.38",
     "@types/moment": "^2.13.0",
```
Comment (illegalprime) [inline package.json]: This should be removed:
```
npm WARN deprecated @types/moment@2.13.0: This is a stub types definition for Moment (https://github.com/moment/moment). Moment provides its own type definitions, so you don't need @types/moment installed!
```

File: package.json [hunk 2]
```diff
@@ -15,13 +15,13 @@
   },
   "dependencies": {
     "body-parser": "^1.17.1",
-    "cheerio": "^0.22.0",
```
Comment (illegalprime) [inline package.json]: `cheerio` is still required in some places:
```
$ git grep cheerio
test/api.ts:import * as cheerio from "cheerio";
test/api.ts:                            let $ = cheerio.load(response.text);
```

File: package.json [hunk 3]
```diff
@@ -32,11 +32,11 @@
   "devDependencies": {
     "@types/body-parser": "0.0.33",
     "@types/chai": "^3.4.34",
-    "@types/cheerio": "^0.17.31",
```
Comment (illegalprime) [inline package.json]: If `cheerio` is still needed, this is obviously still needed, just a reminder comment.

File: package.json (hunk 2 above)
Comment (petschekr) [inline package.json]: Thanks for catching this!

File: package.json (hunk 1 above)
Comment (petschekr) [inline package.json]: :+1:

**Comment by petschekr** [comment]
@illegalprime Thanks for letting me know about those. I amended the changes in fb6837882163b7ffc61078ac1e1d8b56d6df1e26

**Pull request merged**